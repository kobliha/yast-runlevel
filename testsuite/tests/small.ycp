
/**
 * tests for small functions that just call SCR::Read...

 serviceStatus
 runInitScript
 saveDefaultRunlevel
 readDefaultRunlevel
 readCurrentRunlevel

 */
{
    include "runlevel/services.ycp";
    include "testsuite.ycp";
    
    any default_runlevel = nil;
    any default_runlevel_orig = nil;
    any current_runlevel = nil;
    
    TEST (``(readCurrentRunlevel ()), [], 1);
    DUMP (current_runlevel);
    
    TEST (``(readDefaultRunlevel ()), [], 2);
    DUMP (default_runlevel);
    DUMP (default_runlevel_orig);

    TEST (``(saveDefaultRunlevel ("new_default")), [], 0);

    TEST (``(runInitScript ("script", "start")), [], 0);
    TEST (``(serviceStatus ("script")), [], 0);
}
