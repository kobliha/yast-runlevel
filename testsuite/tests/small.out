Read	.init.scripts.current_runlevel 1
Return	nil
Dump	1
Read	.init.scripts.default_runlevel 2
Return	nil
Dump	2
Dump	2
Write	.init.scripts.default_runlevel "new_default" true
Return	true
Execute	.target.bash_output "TERM=raw /etc/init.d/script start" 0
Return	0
Execute	.target.bash "TERM=raw /etc/init.d/script status" 0
Return	0
